autogpt/config/ai_config.py

```diff
@@ -46,71 +49,157 @@ def __init__(
             ai_role (str): The description of the AI's role.
             ai_goals (list): The list of objectives the AI is supposed to complete.
             api_budget (float): The maximum dollar value for API calls (0.0 means infinite)
+            plugins (list): The list of objectives the AI is supposed to complete.
         Returns:
             None
         """
         if ai_goals is None:
             ai_goals = []
+        if plugins is None:
+            plugins = []
         self.ai_name = ai_name
         self.ai_role = ai_role
         self.ai_goals = ai_goals
         self.api_budget = api_budget
+        self.plugins = plugins
         self.prompt_generator: PromptGenerator | None = None
         self.command_registry: CommandRegistry | None = None
 
     @staticmethod
-    def load(config_file: str = SAVE_FILE) -> "AIConfig":
+    def load(ai_name: str, config_file: str = SAVE_FILE) -> Optional["AIConfig"]:
         """
-        Returns class object with parameters (ai_name, ai_role, ai_goals, api_budget) loaded from
-          yaml file if yaml file exists,
-        else returns class with no parameters.
+        Load a specific AI configuration from the config file.
 
-        Parameters:
-           config_file (int): The path to the config yaml file.
-             DEFAULT: "../ai_settings.yaml"
+        Args:
+            ai_name (str): The name of the AI configuration to load.
+            config_file (str): The path to the configuration file.
 
         Returns:
-            cls (object): An instance of given cls object
+            AIConfig: The loaded AI configuration, or None if no such configuration exists.
         """
 
+        all_configs = AIConfig.load_all(config_file)  # type: ignore
+
+        if ai_name in all_configs:
+            return all_configs[ai_name]
+        else:
+            return None
+
+    @staticmethod
+    def load_all(config_file: str = SAVE_FILE) -> Dict[str, "AIConfig"]:
         try:
             with open(config_file, encoding="utf-8") as file:
-                config_params = yaml.load(file, Loader=yaml.FullLoader) or {}
+                all_configs: Dict[str, Any] = yaml.safe_load(file) or {}
         except FileNotFoundError:
-            config_params = {}
-
-        ai_name = config_params.get("ai_name", "")
-        ai_role = config_params.get("ai_role", "")
-        ai_goals = [
-            str(goal).strip("{}").replace("'", "").replace('"', "")
-            if isinstance(goal, dict)
-            else str(goal)
-            for goal in config_params.get("ai_goals", [])
-        ]
-        api_budget = config_params.get("api_budget", 0.0)
-        # type: Type[AIConfig]
-        return AIConfig(ai_name, ai_role, ai_goals, api_budget)
-
-    def save(self, config_file: str = SAVE_FILE) -> None:
+            all_configs = {}
+
+        configs = all_configs.get("configs", {}) or {}
+
+        ai_configs = {}
+        for ai_name, config_params in configs.items():
```

Comment:
Would this be a breaking change with existing `ai_settings.yaml` files? If possible a backwards-compatible change would be good. We've already had so many breaking changes lately.